src/OrchardCore/OrchardCore.Infrastructure/Documents/DefaultDocumentSerializer.cs

```diff
@@ -57,37 +62,23 @@ internal static bool IsCompressed(byte[] data)
         return false;
     }
 
-    internal static byte[] Compress(byte[] data)
+    internal static int Compress(byte[] data, Stream output)
     {
         using var input = new MemoryStream(data);
-        using var output = new MemoryStream();
-        using (var gzip = new GZipStream(output, CompressionMode.Compress))
-        {
-            input.CopyTo(gzip);
-        }
+        using var gZip = new GZipStream(output, CompressionMode.Compress);
 
-        if (output.TryGetBuffer(out var buffer))
-        {
-            return buffer.Array;
-        }
+        input.CopyTo(gZip);
 
-        return output.ToArray();
+        return (int)gZip.Length;
```

Comment:
I don't think it's necessary or useful. The stream has a `Length` property.
We talked about need to know how much was written for the `GetSpan` case.